upup/models/cloudup/_aws/topologies/_topology_private/network.yaml

```diff
@@ -0,0 +1,194 @@
+# ---------------------------------------------------------------
+#
+# Private Network Topology in AWS
+#
+# Inspired by https://github.com/kubernetes/kops/issues/428
+#
+# This will handle deploying k8s instance into a private subnet.
+#
+# Where a private subnet means:
+# A subnet doesn't have a route to the Internet gateway.
+#
+# Utility Subnet - A subnet that is used to bridge a private
+# subnet to a public subnet with a NGW
+#
+# ---------------------------------------------------------------
+
+
+
+
+
+
+# ---------------------------------------------------------------
+# VPC
+#
+# This is a single VPC that will hold all networking componets for
+# a k8s cluster
+#
+# ---------------------------------------------------------------
+vpc/{{ ClusterName }}:
+  id: {{ .NetworkID }}
+  shared: {{ SharedVPC }}
+  cidr: {{ .NetworkCIDR }}
+  enableDnsSupport: true
+  enableDnsHostnames: true
+
+# ---------------------------------------------------------------
+# DHCP
+#
+# If this is not a shared VPC
+# (There is more than one availability zone for this cluster)
+#
+# Also add support for us-east-1
+# ---------------------------------------------------------------
+{{ if not SharedVPC }}
+dhcpOptions/{{ ClusterName }}:
+  domainNameServers: AmazonProvidedDNS
+{{ if eq Region "us-east-1" }}
```

Comment:
1. "us-east-1" is not going away any time soon to be honest.. Also I don't think a find/replace in a directory is that challenging of a task for any developer in this codebase. Basically these yaml models have no concept of global variables or constants. Implementing something like `$us-east-1="us-east-1"` doesn't make very much sense. We could stick something in GO that could serve as a single switch - but again I think coding it in (and supporting it) would take longer/more effort than a good ol `s/us-east-1/new-region` on the very rare off chance we would need to change this.

2. The line in question is checking if the region is "us-east-1" because that region (and only that region) handles private DNS differently than the others. If we needed to change the value here from anything other than "us-east-1" I think it's safe to say the if statement could be removed all together :)

3. This was in place long before this PR. Check it out live in the master branch [here](https://github.com/kubernetes/kops/blob/master/upup/models/cloudup/_aws/network.yaml#L12). Abstracting strings into constants in the models is another topic (that could and should be brought up) separately, and is probably outside the scope of this PR.

4. Models are probably going away soon anyway https://github.com/kubernetes/kops/issues/744